branch_test/copy_phrase_simulation.py

```diff
@@ -0,0 +1,186 @@
+    raise NotImplementedError("Language model option is not implemented!")
+
+conjugator = EvidenceFusion(evidence_names, len_dist=len_alphabet)
+
+# load data (x) and labels (y)
+tmp = sio.loadmat(path + filename)
+x = tmp['scores']
+y = tmp['trialTargetness']
+
+# remove data outliers with a threshold (caveman solution)
+sc_threshold = 10000
+y = y[x > -sc_threshold]
+x = x[x > -sc_threshold]
+y = y[x < sc_threshold]
+x = x[x < sc_threshold]
+
+# shift the positive class
+x[y == 1] += delta
+
+for idx_phrase in range(len(list_phrase)):
+
+    # Initialize the synthetic user
+    pre_phrase = list_phrase[idx_phrase]
+    phrase = target_phrase[idx_phrase]
+    synth = BinaryRSVPSynth(x=x, y=y, phrase=phrase, alp=alphabet,
+                            erase_command=BACKSPACE_CHAR)
+
+    # Initialize the decision maker
+    decision_maker = DecisionMaker(state='', alphabet=alphabet,
+                                   stimuli_agent=stimuli_agent,
+                                   stopping_evaluator=stopping_criteria)
+
+    # this is the dummy eeg_modelling part
+    # currently it assumes the generative model and user EEG match perfectly
+    bandwidth = 1.06 * min(np.std(x),
+                           iqr(x) / 1.34) * np.power(x.shape[0], -0.2)
+    classes = np.unique(y)
+    cls_dep_x = [x[np.where(y == classes[i])] for i in range(len(classes))]
+
+    dist_evidence = []
+    for i in range(len(classes)):
+        dist_evidence.append(KernelDensity(bandwidth=bandwidth))
+
+        dat = np.expand_dims(cls_dep_x[i], axis=1)
+        dist_evidence[i].fit(dat)
+
+    for idx in tqdm(range(max_num_iter)):
+
+        decision_maker.reset(state=pre_phrase)
+        synth.reset()
+        synth.update_state(decision_maker.displayed_state)
+
+        # Continue till the user successfully completes the task
+        while decision_maker.displayed_state != phrase:
+
+            # get prior information from language model
+            tmp_displayed_state = "_" + decision_maker.displayed_state
+            # use language model if language model flag is on
+            if lm_flag:
+                # If language model is present update the prior
+                # Initialize the backspace probability
+                idx_final_space = len(tmp_displayed_state) - \
+                                  list(tmp_displayed_state)[
+                                  ::-1].index("_") - 1
+                l_model.set_reset(
+                    tmp_displayed_state[idx_final_space + 1:])
+                l_model.set_reset(
+                    decision_maker.displayed_state.replace('_', ' '))
+                lm_prior = l_model.get_prob()
+            else:
+                lm_prior = np.ones(len_alphabet) / len_alphabet
+
+            prob = conjugator.update_and_fuse({evidence_names[0]: lm_prior})
+            prob_ = np.array([i for i in prob])
+
+            # Update the decision maker
+            d, sti = decision_maker.decide(prob_)
```

Comment:
unless doing math and needing x, y, theta, etc, single-letter variables are highly discouraged. `decision, stimuli` would be more descriptive here and help future readers of your code.

Reply:
parameter names are updated as `decision, stimuli`